Slack.Automation/Promact.Core.Repository/MailSettingRepository/MailSettingRepository.cs

```diff
@@ -0,0 +1,179 @@
+            if (mailSettingDetails != null)
+            {
+                List<string> listOfTo = new List<string>();
+                List<string> listOfCC = new List<string>();
+                mailSetting = _mapper.Map<MailSetting, MailSettingAC>(mailSettingDetails);
+                var mailSettingMappingList = (await _mailSettingMappingDataRepository.FetchAsync(x => x.MailSettingId == mailSettingDetails.Id)).ToList();
+                var To = mailSettingMappingList.FindAll(x => x.IsTo == true);
+                var CC = mailSettingMappingList.FindAll(x => x.IsTo == false);
+                foreach (var to in To)
+                {
+                    if (to.GroupId == null)
+                        listOfTo.Add(to.Email);
+                    else
+                        listOfTo.Add((await _groupDataRepository.FirstOrDefaultAsync(x => x.Id == to.GroupId)).Name);
+                }
+                foreach (var cc in CC)
+                {
+                    if (cc.GroupId == null)
+                        listOfCC.Add(cc.Email);
+                    else
+                        listOfCC.Add((await _groupDataRepository.FirstOrDefaultAsync(x => x.Id == cc.GroupId)).Name);
+                }
+                mailSetting.To = listOfTo;
+                mailSetting.CC = listOfCC;
+            }
+            return mailSetting;
+        }
+
+        /// <summary>
+        /// Method to add mail setting configuration
+        /// </summary>
+        /// <param name="mailSettingAC">mail setting details</param>
+        public async Task AddMailSettingAsync(MailSettingAC mailSettingAC)
+        {
+            var mailSetting = _mapper.Map<MailSettingAC, MailSetting>(mailSettingAC);
+            mailSetting.CreatedOn = DateTime.UtcNow;
+            _mailSettingDataRepository.Insert(mailSetting);
+            await _mailSettingDataRepository.SaveChangesAsync();
+            foreach (var to in mailSettingAC.To)
+            {
+                var mailSettingMapping = await MailSettingMappingGeneratorAsync(to, true, mailSetting.Id);
+                mailSettingMapping.CreatedOn = DateTime.UtcNow;
+                _mailSettingMappingDataRepository.Insert(mailSettingMapping);
+                await _mailSettingMappingDataRepository.SaveChangesAsync();
+            }
+            foreach (var cc in mailSettingAC.CC)
+            {
+                var mailSettingMapping = await MailSettingMappingGeneratorAsync(cc, false, mailSetting.Id);
+                mailSettingMapping.CreatedOn = DateTime.UtcNow;
+                _mailSettingMappingDataRepository.Insert(mailSettingMapping);
+                await _mailSettingMappingDataRepository.SaveChangesAsync();
+            }
+        }
+
+        /// <summary>
+        /// Method to get list of groups from group table
+        /// </summary>
+        /// <returns>list of group</returns>
+        public async Task<List<string>> GetListOfGroupsAsync()
+        {
```

Comment:
code must be optimized  and update method name (It should be "GetListOfGroupNameAsync")